apis/network/v1/gkenetworkparamset_types.go

```diff
@@ -126,16 +126,27 @@ const (
 	DPDKUnsupported GKENetworkParamSetConditionReason = "DPDKUnsupported"
 	// GNPReady indicates that this GNP resource has been successfully validated and Ready=True
 	GNPReady GKENetworkParamSetConditionReason = "GNPReady"
+	// NetworkAttachmentInvalid indicates that the specified network attachment is invalid.
+	NetworkAttachmentInvalid GKENetworkParamSetConditionReason = "NetworkAttachmentInvalid"
+	// GNPConfigInvalidArgument indicates that the GNP config has an invalid combination of fields specified.
+	GNPConfigInvalidArgument GKENetworkParamSetConditionReason = "GNPConfigInvalidArgument"
 )
 
 // GNPNetworkParamsReadyConditionReason defines the set of reasons that explains
 // the ParamsReady condition on the referencing Network resource.
 type GNPNetworkParamsReadyConditionReason string
 
 const (
-	// L3SecondaryMissing indicates that the L3 type Network resource is
-	// referencing a GKENetworkParamSet with secondary range unspecified.
+	// DEPRECATED. Use L3GNPInvalid instead. L3SecondaryMissing indicates
```

Comment:
Why deprecate and create less specific error?

Reply:
L3 networks with network attachments do not have secondary ranges. When using these errors in validation, an error string with specific information will still be returned.

Reply:
Dont think that matters. You just need to adjust on when this error is set i.e. when there is no NetworkAttachment defined.

Reply:
I'll update these ready conditions to have specific errors rather than "L3GNPInvalid", which will include keeping "L3SecondaryMissing"

Reply:
Updated -- there was actually no need to change the network x GNP conditions except for adding "NetworkAttachmentUnsupported".